### PR TITLE
use compensation from reference implementation

### PR DESCRIPTION
**Pull request by el-han**
This PR is a proposed fix for #78 

It uses the reference implementation of the compensation algorithm from the datasheet. This implementation uses division operations instead of bit shifts.

### CODE REVIEW

**Comment by caternuson** [comment]
Current code seems to match section 4.2.3 of datasheet:
https://www.bosch-sensortec.com/media/boschsensortec/downloads/datasheets/bst-bme280-ds002.pdf
Can you clarify where you are seeing the implementation you have coded?

**Comment by el-han** [comment]
I used the driver which is provided by Bosch as a reference:
https://github.com/BoschSensortec/BME280_driver

**Comment by caternuson** [comment]
Thanks, so:

 * temperature from [here](https://github.com/BoschSensortec/BME280_driver/blob/3d686a326565b9cadb8507f73d7576ec0cc891f2/bme280.c#L1179)
* pressure from [here](https://github.com/BoschSensortec/BME280_driver/blob/3d686a326565b9cadb8507f73d7576ec0cc891f2/bme280.c#L1263)
* humidity from [here](https://github.com/BoschSensortec/BME280_driver/blob/3d686a326565b9cadb8507f73d7576ec0cc891f2/bme280.c#L1322)

Running the library example `bme280test` with your updated code seems to work OK except for pressure, which looks to be off by a factor of 256 (or 2.56?). Test was run near sea level, so ~14.7 psi -> 101352.9 Pa = 1013.5 hPa

Screenshot from 2020-04-30 08-53-32

**Comment by el-han** [comment]
Aw bummer! Seems like I have overlooked the return line in `readPressure()`. Sorry for that! Yes, the pressure is off by factor 2.56. I will fix that.

File: Adafruit_BME280.cpp [hunk 1]
```diff
@@ -441,61 +437,59 @@ float Adafruit_BME280::readPressure(void) {
...
+  var4 = 1048576 - adc_P;
+  var4 = (((var4 * 2147483648) - var2) * 3125) / var1;
+  var1 = (((int64_t)_bme280_calib.dig_P9) * (var4 / 8192) * (var4 / 8192)) /
+         33554432;
+  var2 = (((int64_t)_bme280_calib.dig_P8) * var4) / 524288;
+  var4 = ((var4 + var1 + var2) / 256) + (((int64_t)_bme280_calib.dig_P7) * 16);
+
+  float P = var4 / 256.0;
+
+  return P;
```
Comment (caternuson) [inline Adafruit_BME280.cpp:460]: No need for `P`. Can just `return (float)var4 / 256.0;`

**Review by caternuson — CHANGES_REQUESTED** [review]

**Comment by caternuson** [comment]
@elli89 Sorry for not getting back to this in a while. See suggested code change above. Also - did you test this, similar to my test above, to make sure the readings come out generally as expected?

File: Adafruit_BME280.cpp (hunk 1 above)
Comment (thijstriemstra) [reply Adafruit_BME280.cpp:460]: ping @elli89

**Comment by caternuson** [comment]
I'm going to go ahead and merge this. The one code suggestion was fairly cosmetic. Hopefully these updates help with #78.

Just tested one more time with latest updates and looks good.
Screenshot from 2021-07-29 15-43-37

**Pull request merged**